solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -0,0 +1,135 @@
+"""
+Provides preprocessing steps to be performed on the timeseries data.
+"""
+
+import pandas as pd
+
+from solarforecastarbiter import datamodel
+from solarforecastarbiter.validation import quality_mapping
+
+
+def apply_validation(data, qfilter, handle_func):
+    """
+    Apply validation steps based on provided filters to the data.
+
+    Parameters
+    ----------
+    qfilter : :class:`solarforecastarbiter.datamodel.QualityFlagFilter`
+    data : :class:`pd.DataFrame`
```

Comment:
code makes it look like this is a dict of DataFrames (observations) or Series (forecasts). Keys are the Observations or Forecasts

Reply:
data, qfilter doc order inconsistent with function signature

Reply:
You're correct. Updated